client/src/components/HistoryViewer/HistoryViewer.js

```diff
@@ -308,97 +257,121 @@ class HistoryViewer extends Component {
    *
    * @returns {HistoryViewerVersionList|null}
    */
-  renderComparisonSelectionList() {
-    const { compare: { versionFrom }, ListComponent } = this.props;
-
+  const renderComparisonSelectionList = () => {
+    const versionFrom = compare.versionFrom || null;
     if (!versionFrom) {
       return null;
     }
-
     const selectionListClasses = classNames(
       'history-viewer__table',
       'history-viewer__table--comparison-selected',
     );
-
     return (
       <ListComponent
         versions={[versionFrom]}
         extraClass={selectionListClasses}
       />
     );
-  }
+  };
 
   /**
    * Renders a list of versions
    *
    * @returns {HistoryViewerVersionList}
    */
-  renderVersionList() {
-    const {
-      isInGridField,
-      ListComponent,
-      CompareWarningComponent,
-      compare,
-      compare: { versionFrom: hasVersionFrom },
-    } = this.props;
-
+  const renderVersionList = () => {
+    const hasVersionFrom = compare.versionFrom || null;
     return (
-      <div className={this.getContainerClasses()}>
+      <div className={getContainerClasses()}>
         <CompareWarningComponent />
 
         <div className={isInGridField ? '' : 'panel panel--padded panel--scrollable'}>
-          {this.renderComparisonSelectionList()}
+          {renderComparisonSelectionList()}
           <ListComponent
-            versions={this.state.versions}
+            versions={versions}
             showHeader={!compare || (compare && !hasVersionFrom)}
-            compareModeAvailable={this.compareModeAvailable()}
+            compareModeAvailable={compareModeAvailable()}
           />
 
           <div className="history-viewer__pagination">
-            {this.renderPagination()}
+            {renderPagination()}
           </div>
         </div>
       </div>
     );
-  }
-
-  renderCompareMode() {
-    const { compare } = this.props;
+  };
 
+  const renderCompareMode = () => {
     if (compare && compare.versionFrom && compare.versionTo) {
-      return this.renderVersionDetail();
+      return renderVersionDetail();
     }
-    return this.renderVersionList();
-  }
-
-  render() {
-    const { graphQLErrors, loading, compare, currentVersion, recordId } = this.props;
+    return renderVersionList();
+  };
 
-    // Do not proceed if there are graphql error.
-    // A toast message will be shown in componentDidMount() or componentDidUpdate()
+  /**
+   * Display a toast if there were any pre-existing graphql errors
```

Comment:
```suggestion
   * Fetch version data on mount
   * Display a toast if there were any pre-existing graphql errors
```
The comment about displaying a toast used to be below `refreshVersionData` and is only for the funcitonality below `refreshVersionData`. If we're documenting this effect as a whole, we need to consider it as a whole.

Reply:
Updated

Reply:
Looks like the "mount" `useEffect` no longer displays a toast? Why was that change made? I am begging you to tell me when you make changes that we haven't discussed and give the reasoning without me having to ask you about it.

Reply:
Not required, if you put a console.log() statement in the replacement for 'componentDidUpdate' you'll see that it's triggered on mount

If the componentDidMount is not removed, then the condition would trigger twice.